lib/elastic_record/errors.rb

```diff
@@ -4,11 +4,27 @@ class Error < StandardError
 
   class ConnectionError < Error
     attr_reader :status_code, :payload
-    def initialize(status_code, message, payload = nil)
-      @status_code = status_code
-      message = "#{message} (for payload '#{payload}')" if payload
-      super(message)
+
+    def initialize(status_code, json_error, json_payload = nil)
+      @status_code  = status_code
+      error_message = build_message_hash(json_error, json_payload).to_json
+
+      super(error_message)
     end
+
+    private
+
+      def build_message_hash(json_error, json_payload)
+        error   = JSON.parse(json_error)
+        payload = JSON.parse(json_payload || '{}')
+
+        error.merge!('payload' => payload)
+      rescue JSON::ParserError
+        {
+          'error'   => json_error,
+          'payload' => json_payload
+        }
```

Comment:
If one is parse-able and the other isn't, no point in not parsing the one that we can parse:

```suggestion
        payload = if json_payload
          begin
            JSON.parse(json_payload)
          rescue JSON::ParserError
            json_payload
          end
        else
          nil
        end
        
        begin
          JSON.parse(json_error).merge(payload:)
        rescue JSON::ParserError
          { error: json_error, payload: }
        end
```